File: hw/jacome/U3/HW23Observer/JS/model/IBM.js
const Stock = require('./Stock');

class IBM extends Stock {
    constructor(symbol, price) {
        super();
        this.symbol = symbol;
        this.price = price;
    }

    getPrice() {
        return this.price;
    }

    setPrice(price) {
        this.price = price;
        this.notifyObservers(price);
    }

    getSymbol() {
        return this.symbol;
    }

    setSymbol(symbol) {
        const oldSymbol = this.symbol;
        this.symbol = symbol;
        this.notifyObservers(`Symbol changed from ${oldSymbol} to ${symbol}`);
    }
}

module.exports = IBM;

File: hw/paillacho/U3/HW23Observer/ObserverPython/src/model/ibm.py
"""
Concrete Subject in the Observer pattern
@author Carlos Paillacho
"""

from .stock import Stock


class IBM(Stock):
    def __init__(self, symbol, price):
        super().__init__()
        self._symbol = symbol
        self._price = price

    def get_price(self):
        return self._price

    def set_price(self, price):
        self._price = price
        self.notify_observers(float(price))

    def get_symbol(self):
        return self._symbol

    def set_symbol(self, symbol):
        old_symbol = self._symbol
        self._symbol = symbol
        self.notify_observers(f"Symbol changed from {old_symbol} to {symbol}")
